File: Week_02/242-valid-anagram.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) {
            return false;
        }

        int n = s.length();
        int[] letters = new int[26];
        for(int i = 0; i < n; i++) {
            letters[s.charAt(i) - 'a'] ++;
            letters[t.charAt(i) - 'a'] --;
        }

        for(int i = 0; i < 26; i++) {
            if(letters[i] != 0) {
                return false;
            }
        }

        return true;
    }
}

File: Week_02/242-valid-anagram.cpp
#include <string>
#include <iostream>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
    if (s.length() != t.length()) {
        return false;
    }
    int letters[26] = {0};
    for(int i = 0; i < s.length(); i++) {
        letters[s[i] - 'a'] ++;
        letters[t[i] - 'a'] --;
    }

    for (auto count : letters)
    {
        if (count != 0)
        {
           return false;
        }
        
    }

    return true;
    }
};

int main(int argc, char const *argv[])
{
    string s = "anagram", t = "nagaram";
    Solution solution;
    bool ret = solution.isAnagram(s, t);
    cout<<ret<<endl;
    return 0;
}
